web/profiles/custom/yalesites_profile/modules/custom/ys_templated_content/src/Modifiers/TemplateModifier.php

```diff
@@ -0,0 +1,228 @@
+   *
+   * @return array
+   *   The content array with images fixed with placeholder.
+   */
+  protected function replaceBrokenImages(array $content_array) : array {
+    foreach ($content_array as $key => $value) {
+      if (is_array($value)) {
+        $content_array[$key] = $this->replaceBrokenImages($value);
+      }
+      elseif ($key == 'uri' && strpos($value, 'public://') !== FALSE) {
+        $path = $value;
+        $path = str_replace('public://', 'sites/default/files/', $path);
+        if (!file_exists($path)) {
+          $content_array[$key] = $this::PLACEHOLDER;
+        }
+      }
+    }
+
+    return $content_array;
+  }
+
+  /**
+   * Replace the UUIDs in the content array.
+   *
+   * Any reference to the main UUID should be replaced so that we reference
+   * the newly created one.
+   *
+   * @param array $content
+   *   The content array.
+   * @param string $uuid
+   *   The UUID to replace.
+   * @param string $newUuid
+   *   The new UUID.
+   *
+   * @return array
+   *   The content array with the UUIDs replaced.
+   */
+  public function replaceUuids($content, $uuid, $newUuid) {
+    if (array_key_exists('uuid', $content) && $content['uuid'] === $uuid) {
+      $content['uuid'] = $newUuid;
+    }
+    // Find any other element that has the original UUID passed and replace it
+    // with the new one.
+    foreach ($content as $key => $value) {
+      if (is_array($value)) {
+        $content[$key] = $this->replaceUuids($value, $uuid, $newUuid);
+      }
+      elseif (is_string($value) && strpos($value, 'entity:node/') !== FALSE) {
+        $content[$key] = str_replace('entity:node/' . $uuid, 'entity:node/' . $newUuid, $value);
+      }
+      elseif (is_string($value) && strpos($value, $uuid) !== FALSE) {
+        $content[$key] = str_replace($uuid, $newUuid, $value);
+      }
+    }
+
+    return $content;
+  }
+
+  /**
+   * Generate a unique alias using the current date/time.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias (original alias with date/time).
+   */
+  protected function generateUniqueAliasWithDate($alias) {
+    $date = date('Y-m-d-H-i-s');
+    $alias .= '-' . $date;
+
+    return $alias;
+  }
+
+  /**
+   * Generate a unique alias with a sequential number.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias (original alias with sequential number).
+   */
+  protected function generateUniqueAliasWithSequentialNumber($alias) {
+    $aliasNumber = 1;
+    $newAlias = $alias . '-' . $aliasNumber;
+
+    while ($this->pathAliasRepository->lookupByAlias($newAlias, 'en')) {
+      $aliasNumber++;
+      $newAlias = $alias . '-' . $aliasNumber;
+    }
+
+    return $newAlias;
+  }
+
+  /**
+   * Generate a unique alias.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias.
+   */
+  protected function generateUniqueAlias($alias) {
```

Comment:
Food for thought. You are recreating logic that Drupal should do for you. If you remove the `url` line from the template, then you can delete all of these alias methods as the pathauto module will just fill in the details for you.

Of course relying on manual edits is never a good long term strategy. Is it better to sanitize the template to remove the 'url' setting? I like the idea of removing all of these alias function if possible.